lib/modem_info/modem_info.c

```diff
@@ -720,6 +720,119 @@ int modem_info_rsrp_register(rsrp_cb_t cb)
 	return 0;
 }
 
+/* FW UUID is 36 characters: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX */
+#define FW_UUID_SIZE 37
+
+int modem_info_get_fw_uuid(char *buf, size_t buf_size)
+{
+	int ret;
+
+	if (buf_size < FW_UUID_SIZE) {
+		return -EINVAL;
+	}
+
+	ret = nrf_modem_at_scanf("AT%XMODEMUUID",
+				 "%%XMODEMUUID: %" STRINGIFY(FW_UUID_SIZE) "[^\r\n]",
+				 buf);
+	if (ret != 1) {
+		LOG_ERR("Could not get FW ID, error: %d", ret);
+		return -EIO;
+	}
+	return 0;
+}
+
+/* The "short software identification" has no specified size limit. */
+#define FW_VERSION_SIZE 32
+
+int modem_info_get_fw_version(char *buf, size_t buf_size)
+{
+	int ret;
+
+	if (buf_size < FW_VERSION_SIZE) {
+		return -EINVAL;
+	}
+
+	ret = nrf_modem_at_scanf("AT%SHORTSWVER",
+				 "%%SHORTSWVER: %" STRINGIFY(FW_VERSION_SIZE) "[^\r\n]",
```

Comment:
Hmm, how would the application know then that the buffer has to be at least 32 bytes? Especially, that the example FW version in the API docs is much shorter. Perhaps `buf_size` should be used directly as a read width here?

Reply:
Thanks for the hint! I didn't know scanf had this option.